src/GafferSceneUI/LightPositionTool.cpp

```diff
@@ -0,0 +1,969 @@
+	{
+		return;
+	}
+	const Selection s = selection().back();
+
+	const V3f newP = shadowSourcePosition( shadowPivot, shadowTarget, pivotDistance );
+
+	// See `RotateTool::buttonPress()` for a description of why we use this relatively
+	// elaborate orientation calculation.
+
+	Context::Scope scopedContext( s.context() );
+
+	const M44f localTransform = s.scene()->transform( s.path() );
+
+	const M44f orientationMatrix = shadowSourceOrientation( s, shadowPivot, shadowTarget );
+
+	V3f originalRotation;
+	extractEulerXYZ( localTransform, originalRotation );
+	const M44f originalRotationMatrix = M44f().rotate( originalRotation );
+
+	const M44f relativeMatrix = originalRotationMatrix.inverse() * orientationMatrix;
+
+	V3f relativeRotation;
+	extractEulerXYZ( relativeMatrix, relativeRotation );
+
+	const V3f p = V3f( 0 ) * s.orientedTransform( Orientation::World );
+	const V3f offset = newP - p;
+
+	TranslationRotation( s ).apply( offset, relativeRotation );
+}
+
+bool LightPositionTool::affectsHandles( const Gaffer::Plug *input ) const
+{
+	if( TransformTool::affectsHandles( input ) )
+	{
+		return true;
+	}
+
+	return input == scenePlug()->transformPlug();
+}
+
+void LightPositionTool::updateHandles( float rasterScale )
+{
+	Selection s = selection().back();
+
+	handles()->setTransform( s.orientedTransform( Orientation::Local ) );
+
+	m_shadowHandle->setEnabled( selection().size() == 1 && TranslationRotation( s ).canApply() );
+
+	m_shadowHandle->setRasterScale( 0 );
+
+	auto shadowHandle = static_cast<ShadowHandle *>( m_shadowHandle.get() );
+
+	m_key = MurmurHash().append( s.path() );
+
+	std::optional<V3f> shadowPivot = getShadowPivot();
+	std::optional<V3f> shadowTarget = getShadowTarget();
+
+	shadowHandle->setShadowPivot( shadowPivot );
+	shadowHandle->setShadowTarget( shadowTarget );
+
+	if( !shadowPivot || !shadowTarget )
+	{
+		return;
+	}
+
+	// The user can control the distance along the line from shadow target
+	// to pivot, and the rotation around the Z-axis. Any variance from those
+	// contraints invalidates the stored shadow parameters.
+
+	Context::Scope scopedContext( s.context() );
+
+	const M44f worldTransform = s.scene()->fullTransform( s.path() );
+	const V3f p = worldTransform.translation();
+	const Line3f handleLine( shadowTarget.value(), shadowPivot.value() );
+
+	V3f direction;
+	worldTransform.multDirMatrix( V3f( 0, 0, -1.f ), direction );
+
+	const V3f handleDir = ( shadowTarget.value() - shadowPivot.value() ).normalized();
+
+	if(
+		!direction.equalWithAbsError( ( shadowTarget.value() - shadowPivot.value() ).normalized(), 1e-4 ) ||
+		handleLine.distanceTo( p ) > shadowHandle->getPivotDistance().value() * 1e-4
+	)
+	{
+		shadowHandle->setShadowPivot( std::nullopt );
+		shadowHandle->setShadowTarget( std::nullopt );
+		shadowHandle->setPivotDistance( std::nullopt );
+
+		m_shadowTargetMap.erase( m_key );
+		m_shadowPivotMap.erase( m_key );
+		m_shadowPivotDistanceMap.erase( m_key );
+	}
```

Comment:
It doesn't seem possible to use the LightPositionTool when there's a Group transforming the light downstream. In the script below you can see this by focussing the Group and then trying to use the tool on `/group/spotLight`. I'm not 100% sure this chunk of code is the culprit, but I suspect so - it seems very much like the pivot/target are getting rejected as soon as they've been set.

This sort of "edit above a downstream transform" thing was very hard to get right for the other tool types - I'd suggest adding some test coverage to give us a safety net.

<details>

```
import Gaffer
import GafferArnold
import GafferScene
import imath

Gaffer.Metadata.registerValue( parent, "serialiser:milestoneVersion", 1, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:majorVersion", 3, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:minorVersion", 8, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:patchVersion", 0, persistent=False )

__children = {}

__children["Cube"] = GafferScene.Cube( "Cube" )
parent.addChild( __children["Cube"] )
__children["Cube"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Plane"] = GafferScene.Plane( "Plane" )
parent.addChild( __children["Plane"] )
__children["Plane"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Parent"] = GafferScene.Parent( "Parent" )
parent.addChild( __children["Parent"] )
__children["Parent"]["children"].addChild( GafferScene.ScenePlug( "child1", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Parent"]["children"].addChild( GafferScene.ScenePlug( "child2", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Parent"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["SpotLight"] = GafferArnold.ArnoldLight( "SpotLight" )
parent.addChild( __children["SpotLight"] )
__children["SpotLight"].loadShader( "spot_light" )
__children["SpotLight"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"] = GafferScene.Group( "Group" )
parent.addChild( __children["Group"] )
__children["Group"]["in"].addChild( GafferScene.ScenePlug( "in1", flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Group"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Cube"]["transform"]["scale"].setValue( imath.V3f( 1, 5.59656191, 1 ) )
__children["Cube"]["__uiPosition"].setValue( imath.V2f( -5.08920288, 3.61759186 ) )
__children["Plane"]["transform"]["rotate"].setValue( imath.V3f( -90, 0, 0 ) )
__children["Plane"]["transform"]["scale"].setValue( imath.V3f( 155.261353, 155.261353, 155.261353 ) )
__children["Plane"]["__uiPosition"].setValue( imath.V2f( 7.91079521, 3.61778164 ) )
__children["Parent"]["in"].setInput( __children["Cube"]["out"] )
__children["Parent"]["parent"].setValue( '/' )
__children["Parent"]["children"][0].setInput( __children["Plane"]["out"] )
__children["Parent"]["children"][1].setInput( __children["SpotLight"]["out"] )
__children["Parent"]["__uiPosition"].setValue( imath.V2f( 8.41027451, -4.546875 ) )
__children["SpotLight"]["name"].setValue( 'spotLight' )
__children["SpotLight"]["transform"]["translate"].setValue( imath.V3f( -6.37608194, 3.91808772, 0.26422596 ) )
__children["SpotLight"]["transform"]["rotate"].setValue( imath.V3f( -114.534935, -95.2462158, 103.170578 ) )
__children["SpotLight"]["parameters"]["exposure"].setValue( 7.0 )
__children["SpotLight"]["__uiPosition"].setValue( imath.V2f( 20.9107952, 3.61795092 ) )
__children["Group"]["in"][0].setInput( __children["Parent"]["out"] )
__children["Group"]["transform"]["rotate"].setValue( imath.V3f( 0, 90, 0 ) )
__children["Group"]["__uiPosition"].setValue( imath.V2f( 9.9087038, -12.7127008 ) )


del __children
```


</details>

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/5558/commits/7a3cf3b4f419b81929e0ac68856f01e4c3bc8594 and https://github.com/GafferHQ/gaffer/pull/5558/commits/2df67e588fdefb72a6188c5a94f7b1d7bb5b3759